v2/apiserver/internal/core/events.go

```diff
@@ -403,13 +417,21 @@ func (e *eventsService) createSingleEvent(
 
 	event.ID = uuid.NewV4().String()
 
+	jobs := []Job{}
 	workerSpec := project.Spec.WorkerTemplate
+	// If the event is a retry of another, defer to the Worker.Spec on the event
+	// itself.  Retain the Jobs slice as well.
+	if event.Labels != nil && event.Labels[RetryLabelKey] != "" {
+		workerSpec = event.Worker.Spec
+		jobs = event.Worker.Jobs
+	}
```

Comment:
Don't we want to be more selective about which jobs we copy over? I think, offline, we had discussed copying over jobs/status only for jobs that succeeded _and_ didn't use the shared workspace. Anything else shouldn't be inherited because it needs to be re-run.

Reply:
Yup, good call.  Updated.